### ISSUE TITLE
Helm chart - Setting settings.clusterName forces settings.itemsPerPage to 0 if not set explicitly

### ISSUE THREAD

**Issue by mlbiam** [kubernetes/dashboard#7445]
### What happened?

When deploying the dashboard into a vCluster, where the metrics services aren't available, the dashboard is unable to list any resources.  
![Screen Shot 2022-09-12 at 11 08 43 AM](https://user-images.githubusercontent.com/8249283/189690141-06ab0431-da20-4c96-a583-ee0fda491234.png)

And when listing `Pod`s:

![Screen Shot 2022-09-12 at 11 10 04 AM](https://user-images.githubusercontent.com/8249283/189690327-0503cfa8-1996-4ef4-a72f-26c0ab427ca8.png)

In the dashboard `Pod`'s logs - 

```
2022/09/12 15:11:06 [2022-09-12T15:11:06Z] Incoming HTTP/1.1 GET /api/v1/pod/openunison?itemsPerPage=0&page=1&sortBy=d%!C(MISSING)creationTimestamp request from 192.168.2.112: 
2022/09/12 15:11:06 Getting list of all pods in the cluster
2022/09/12 15:11:06 [2022-09-12T15:11:06Z] Outcoming response to 192.168.2.112 with 200 status code
2022/09/12 15:11:06 Internal error occurred: No metric client provided. Skipping metrics.
2022/09/12 15:11:06 Getting pod metrics
2022/09/12 15:11:06 Internal error occurred: No metric client provided. Skipping metrics.
2022/09/12 15:11:06 [2022-09-12T15:11:06Z] Outcoming response to 192.168.2.112 with 200 status code
```

However, if I try to get pods using the same user:

```
k get pods -n openunison
NAME                                             READY   STATUS      RESTARTS   AGE
openunison-operator-6cb7444cc6-jhvqj             1/1     Running     0          9d
openunison-orchestra-8dcc5dbd7-44jzl             1/1     Running     0          9d
kube-oidc-proxy-orchestra-68c66857f9-fqcpz       1/1     Running     0          9d
ouhtml-orchestra-login-portal-7fcb97b996-mzbcx   1/1     Running     0          9d
check-certs-orchestra-27712920-2dzb9             0/1     Completed   0          2d13h
check-certs-orchestra-27714360-2bmf2             0/1     Completed   0          37h
check-certs-orchestra-27715800-ghwqk             0/1     Completed   0          13h
```

The integration with dashboard is via an impersonating reverse proxy.

### What did you expect to happen?

pods should be listed

### How can we reproduce it (as minimally and precisely as possible)?

probably the easiest way: https://www.tremolosecurity.com/post/vcluster-enterprise-authentication

### Anything else we need to know?

The vCluster has no metrics deployed

### What browsers are you seeing the problem on?

Chrome, Safari, Firefox

### Kubernetes Dashboard version

NAME                	NAMESPACE           	REVISION	UPDATED                                	STATUS  	CHART                      	APP VERSION kubernetes-dashboard	kubernetes-dashboard	1       	2022-09-02 18:08:46.044357215 +0000 UTC	deployed	kubernetes-dashboard-5.10.0	2.6.1

### Kubernetes version

1.23

**Comment by floreks** [comment]
The metrics log is basically a warning. Go to settings view and increase items per page number. It looks like it is 0. Possibly you have used some custom way to deploy Dashboard and settings are not available.

**Comment by mlbiam** [comment]
> Go to settings view and increase items per page number. 

yep, this fixed it

> Possibly you have used some custom way to deploy Dashboard and settings are not available.

no, just using the helm chart:

```
kubectl create ns kubernetes-dashboard
helm install kubernetes-dashboard kubernetes-dashboard/kubernetes-dashboard --set settings.clusterName=$VCLUSTER_LABEL -n kubernetes-dashboard
```

**Comment by mlbiam** [comment]
Looks like the issue is that if any of the `settings` options are set, then any that aren't explicitly set are set to 0

**Comment by kiich** [comment]
Just to say we have also ran into this when we set the 
```
  _global: '{"logsAutoRefreshTimeInterval":10,"resourceAutoRefreshTimeInterval":0}'
```
those took effect but it set itemsPerPage to 0 causing the mentioned issue.
It seems the entire `settings` block gets dumped if you set any of them.

**Comment by floreks** [comment]
Fixed by #8708
